flink/v2.1/flink/src/main/java/org/apache/iceberg/flink/data/FlinkParquetReaders.java

```diff
@@ -404,8 +404,10 @@ private static class NanosToTimestampReader
     @Override
     public TimestampData read(TimestampData ignored) {
       long value = readLong();
+      long mills = Math.floorDiv(value, 1_000_000_000L);
+      long nanos = Math.floorMod(value, 1_000_000_000L);
       return TimestampData.fromEpochMillis(
-          Math.floorDiv(value, 1_000_000L), Math.floorMod(value, 1_000_000));
+          mills + Math.floorDiv(nanos, 1_000_000L), (int) Math.floorMod(nanos, 1_000_000L));
```

Comment:
I don't understand this change.
Why are we precalculating seconds? And then again converting to millis?